Sources/Citadel/SFTP/SFTPFile.swift

```diff
@@ -1,34 +1,142 @@
 import NIO
+import Logging
 
-public struct SFTPFile {
-    let handle: ByteBuffer
-    let client: SFTPClient
+/// A "handle" for accessing a file that has been successfully opened on an SFTP server. File handles support
+/// reading (if opened with read access) and writing/appending (if opened with write/append access).
+public class SFTPFile {
+    /// A typealias to clarify when a buffer is being used as a file handle.
+    ///
+    /// This should probably be a `struct` wrapping a buffer for stronger type safety.
+    public typealias SFTPFileHandle = ByteBuffer
     
-    public func readChunk(from offset: UInt64 = 0, length: UInt32) -> EventLoopFuture<ByteBuffer> {
-        client.readFile(handle: handle, offset: offset, length: length)
+    /// Indicates whether the file's handle was still valid at the time the getter was called.
+    public private(set) var isActive: Bool
```

Comment:
Does something more explicit such as `hasOpenFileHandle` make more sense?

Reply:
I originally chose `isActive` for consistency with NIO's usage, if not for that I'd agree.